pkg/reconciler/managed/reconciler.go

```diff
@@ -545,10 +545,10 @@ func (r *Reconciler) Reconcile(_ context.Context, req reconcile.Request) (reconc
 		// If this is the first time we encounter this issue we'll be requeued
 		// implicitly when we update our status with the new error condition.
 		// If not, we want to try again after a short wait.
-		log.Debug("Cannot initialize managed resource", "error", err, "requeue-after", time.Now().Add(r.shortWait))
+		log.Debug("Cannot initialize managed resource", "error", err)
 		record.Event(managed, event.Warning(reasonCannotInitialize, err))
 		managed.SetConditions(xpv1.ReconcileError(err))
-		return reconcile.Result{RequeueAfter: r.shortWait}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
+		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
```

Comment:
Would we be able to just `return reconcile.Result, err` in these situations if we deprecated the `Synced` status per https://github.com/crossplane/crossplane-runtime/issues/198? Just curious - we don't need to bring that into scope with this PR.

Reply:
I believe we still need to call the status update in some scenarios (such as #242). One thing we could do it concatenate the `err` from the method that was called with the status update. I played around with this a bit initially though and it felt pretty sloppy.

Reply:
Ah, that's true. We also need to assume that at the `Observe` method (and possibly also `Create` and `Update`) could be mutating the managed resource's status.

That said, perhaps it's acceptable for us to only persist these status updates in the happy path? i.e. In the return statements that follow any place we'd currently log "Successfully Xed the external resource". When we hit an error case we'd return the error without persisting the managed resource's status. When we didn't hit an error we'd call `r.client.Status().Update()` as we do today, and return that error if there was one.

Without having thought through it too much I feel like this is probably acceptable? Managed resource viewers should be able to tell that we're trying to create or delete a managed resource because we'll be emitting warning events like `CannotCreateExternalResource`. Any status fields that would be set by the `ExternalClient` methods would presumably just get set once the error cleared and we were able to proceed.

Reply:
> perhaps it's acceptable for us to only persist these status updates in the happy path? i.e. In the return statements that follow any place we'd currently log "Successfully Xed the external resource". When we hit an error case we'd return the error without persisting the managed resource's status.

Would that result in errors not appearing in conditions, hence shown only as event?

Reply:
@muvaf yep, I don't think we can do this currently